### PR TITLE
Expose Prefix and Suffix types

### PR DESCRIPTION
**Pull request by dflemstr**
This is useful if you are integrating with a library that only lets you write to a writer via callbacks.  Like:

```rust
impl somelib::Callback for MyANSITermCallback {
  fn before(&self, f: &mut fmt::Formatter) -> fmt::Result {
    write!(f, "{}", self.style.prefix())
  }

  fn after(&self, f: &mut fmt::Formatter) -> fmt::Result {
    write!(f, "{}", self.style.suffix())
  }
}
```

I will improve the rustdoc if people think the general API is sane.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -686,27 +757,27 @@ enum Difference {
 pub struct ANSIStrings<'a>(pub &'a [ANSIString<'a>]);
 
 impl<'a> fmt::Display for ANSIStrings<'a> {
-	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-	    let first = match self.0.first() {
-	        None => return Ok(()),
-	        Some(f) => f,
-	    };
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
```
Comment (ogham) [inline src/lib.rs]: I totally thought I de-tabbed everything ages ago!

**Comment by ogham** [comment]
Looks good. I didn't understand what this was for at first, but now I get it!

I merged another PR first though so I'll have to merge it manually...

**Comment by ogham** [comment]
I published this as v0.9.0. Mind giving this a test and seeing if it works for you?

**Comment by dflemstr** [comment]
@ogham I was using the branch before already! Works well!

**Pull request merged**